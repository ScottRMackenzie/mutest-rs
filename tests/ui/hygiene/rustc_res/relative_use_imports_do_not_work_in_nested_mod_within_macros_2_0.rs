//@ fail
//@ stderr

#![feature(decl_macro)]

#![allow(unused)]

macro m() {
    mod def {
        pub fn f() {}

        pub mod inner {
            pub mod within {
                pub fn g() {}
            }

            pub macro m() {
                pub mod within_macro {
                    use super::super::f;
                    use super::within::*;
                }
            }
        }
    }

    def::inner::m!();
}

m!();
