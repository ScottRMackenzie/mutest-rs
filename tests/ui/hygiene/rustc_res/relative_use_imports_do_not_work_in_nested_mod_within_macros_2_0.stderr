error[E0432]: unresolved import `super::super::f`
  --> tests/ui/hygiene/rustc_res/relative_use_imports_do_not_work_in_nested_mod_within_macros_2_0.rs:20:25
   |
20 |                     use super::super::f;
   |                         ^^^^^^^^^^^^^^-
   |                         |             |
   |                         |             help: a similar name exists in the module: `m`
   |                         no `f` in `def::inner`
...
30 | m!();
   | ---- in this macro invocation
   |
   = note: this error originates in the macro `def::inner::m` which comes from the expansion of the macro `m` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
